source/extensions/load_balancing_policies/common/load_balancer_impl.cc

```diff
@@ -637,18 +640,44 @@ absl::FixedArray<ZoneAwareLoadBalancerBase::LocalityPercentages>
 ZoneAwareLoadBalancerBase::calculateLocalityPercentages(
     const HostsPerLocality& local_hosts_per_locality,
     const HostsPerLocality& upstream_hosts_per_locality) {
-  uint64_t total_local_hosts = 0;
-  std::map<envoy::config::core::v3::Locality, uint64_t, LocalityLess> local_counts;
+  std::map<envoy::config::core::v3::Locality, uint64_t, LocalityLess> local_weights;
+  std::map<envoy::config::core::v3::Locality, uint64_t, LocalityLess> upstream_weights;
+  uint64_t total_local_weight = 0;
   for (const auto& locality_hosts : local_hosts_per_locality.get()) {
-    total_local_hosts += locality_hosts.size();
+    uint64_t locality_weight = 0;
+    for (const auto& host : locality_hosts) {
+      // If use_host_weight_ is true, we use the host's weight to calculate the locality percentage.
+      // Otherwise, we count the number of hosts in the locality.
+      if (use_host_weight_) {
```

Comment:
I was just walking around. And I'm not related to the Envoy developers in any way.
But still, is it possible to move `use_host_weight_ ` outside of the `for` loop?
Consider the default case for many users when `use_host_weight_ ` is set to false. Instead of simple call for `locality_hosts.size();` new code will do N times of `locality_weight += 1;`